packages/spec-dashboard/src/utils/tier-filtering-utils.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Configuration for classifying scenarios into tiers.
+ */
+export interface TierConfig {
+  default: string;
+  tiers: Record<string, string[]>; // only arrays allowed
+}
+
+/**
+ * Compiled tier config for efficient classification.
+ */
+export interface CompiledTierConfig {
+  exact: Map<string, string>;
+  prefixes: { tier: string; prefix: string }[];
+  defaultTier: string;
+}
+
+/**
+ *  Compiles a TierConfig into a CompiledTierConfig for efficient classification.
+ */
+export function compileTierConfig(config: TierConfig): CompiledTierConfig {
+  const exact = new Map<string, string>();
+  const prefixes: { tier: string; prefix: string }[] = [];
+
+  const { tiers, default: defaultTier } = config;
+
+  if (!tiers[defaultTier]) {
+    throw new Error(`Invalid tierConfig.default: "${defaultTier}" not found in tiers`);
+  }
+
+  for (const [tier, patterns] of Object.entries(tiers)) {
+    for (const pattern of patterns) {
+      if (pattern.includes("*")) {
```

Comment:
do you think this is worth it to convert to a prefix vs a regex which would allow `*` in the middle/start?